code/renderers/react/src/mount.ts

```diff
@@ -0,0 +1,11 @@
+import { type StoryContext, type ReactRenderer } from './public-types';
+import type { BaseAnnotations } from '@storybook/types';
+
+export const mount: BaseAnnotations<ReactRenderer>['mount'] =
+  (context: StoryContext) => async (ui) => {
+    if (ui != null) {
+      context.originalStoryFn = () => ui;
```

Comment:
@tmeasday The reason I do it like this, is because this makes sure that any decorator that looks for the `originalStoryFn` will still work with this new `mount` model:

https://github.com/storybookjs/storybook/blob/cb03069173260a6dbe1d7d59add5ddd35394d0ff/code/renderers/html/src/docs/sourceDecorator.ts#L25-L45

My personal feeling is that `context.originalStoryFn` should be renamed to `context.render` in 9.0, and that we should think about `mount` as lazily loading the `context.render` definition.

Reply:
I do think it makes sense to have a render defined as `project` or `component` annotations, and still override it in `play` with `mount`.

When the user defines a render in the story annotation and in mount, then the `render` definition is kind of unused indeed.

```tsx
const Story = {
  render: () => ({component: '<div>foo</div>' })
  play: async ({ mount }) => {
    await mount({ component: '<div>something else</div>' }),
  },
}
```

But I don't think we can recognise that in runtime, maybe at some point, we could recognize that in the ESLint plugin. But my feeling is that it will be pretty obvious to the user that the `render` method gets overriden in such cases.